components/browser/engine-system/src/main/java/mozilla/components/browser/engine/system/SystemEngineSession.kt

```diff
@@ -15,6 +15,7 @@ import java.lang.ref.WeakReference
  */
 class SystemEngineSession : EngineSession() {
     internal var urlToLoad: String? = null
```

Comment:
nit: finding the naming a bit confusing now `urlToLoad` isn't necessarily a URL. Should we rename it? We could also create a data class ScheduledLoad or similar which takes a url, data, and mime type?

Reply:
I wasn't too keen on my original implementation as well since it reduced the test coverage percentage. The data class sounds like a better idea!